packages/forms/src/UIForm/UIForm.container.js

```diff
@@ -143,8 +143,20 @@ export default class UIForm extends React.Component {
 
 	onTrigger(event, payload) {
 		return this.props.onTrigger(event, payload).then(data => {
+			const liveState = this.state.liveState;
 			if (data.errors) {
-				this.setErrors(event, data.errors);
+				let errors = data.errors;
+				if (typeof data.errors === 'function') {
+					errors = data.errors(liveState.errors);
+				}
+				this.setErrors(event, errors);
+			}
+			if (data.properties) {
+				let properties = data.properties;
+				if (typeof data.properties === 'function') {
+					properties = data.properties(liveState.properties);
+				}
+				this.setState(setLiveStateProperties(properties));
```

Comment:
I would have prefer a pacth system here.
The global properties is impossible to merge.
trigger may take time so the properties is clearly not in synch here

Reply:
The patch system is the if just above